src/wallet/wallet.cpp

```diff
@@ -626,6 +626,15 @@ bool CWallet::EncryptWallet(const SecureString& strWalletPassphrase)
 
         Lock();
         Unlock(strWalletPassphrase);
+
+        // if we are using HD, replace the HD master key with a new one
+        if (!hdChain.masterKeyID.IsNull()) {
+            CKey key;
+            key.MakeNewKey(true);
+            if (!SetHDMasterKey(key))
```

Comment:
I think this solution is fine for the moment. The return value of `SetHDMasterKey()` is for future extensions and I think its useful even if its not possible to get a `false` returned with the current implementation.

The exception will already be caught though the RPC/Server logic (`EncryptWallet()` will only be called in the RPC context where we "handle" exceptions).